src/Controller/AbstractController.php

```diff
@@ -74,7 +76,21 @@ private function findLayoutFile(string $file): string
     }
 
     /**
-     * Returns the name of the controller.
+     * Returns the controller name. Name should be converted to "id" case.
+     *
+     * @return string
+     * @example If class named MySiteController method will return my-site
+     * @see Inflector::camel2id()
      */
-    abstract protected function name(): string;
+    protected static function getName(): string
+    {
+        if (static::$controllerName !== null) {
+            return static::$controllerName;
+        }
+
+        $name = preg_replace('/(?:.*\\\)([a-z]+)(controller)/iu', '$1', static::class);
```

Comment:
Does it take namespace into account? What would be name for app Site Controller?

Reply:
I wrote some words about algroithm [here](https://github.com/yiisoft/app/blob/128ba3fb4fcba2f81050a4aa060b1f6615f626fc/src/Controller/AbstractController.php#L79)

Reply:
https://regex101.com/r/VYp9z0/1